### PR TITLE
Enhance InMemoryLedger to use the ValueEnricher

### PR DESCRIPTION
**Pull request by nickchapman-da**
Enhance `InMemoryLedger` to use the `ValueEnricher`.
- Allowing `SandboxServer` to run Engine with the default `transactionNormalization=false`.
- Enrichment is *not* done when transactions are published; the ACS stores un-enriched contracts and txs.
- Enrichment is done only when responding to verbose API queries.

This change advances the state of #6665

### CODE REVIEW

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala [hunk 1]
```diff
@@ -353,12 +367,14 @@ private[sandbox] final class InMemoryLedger(
                 transactionId,
                 transaction,
               )
+          val enrichedCommittedTransaction =
```
Comment (cocreature) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala:370]: i don’t think always enriching is the right approach. We should do this in the translation if the `verbose` flag is set not always.

Comment (nickchapman-da) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala:370]: I did initially try the approach of enriching when returning transactions and contracts form the ledger (conditional on `verbose`), but I couldn't get it to work.

I think the problem is that some of the lookup methods might be dependant on having the in memory store the enriched transaction.

Comment (cocreature) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala:370]: Which lookups do you mean and which error did you get when trying that? Doing the enrichment conditionally on the `verbose` flag is what the SQL backend does so I don’t understand why it doesn’t work here.

Comment (nickchapman-da) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala:370]: discussion moved to slack

Comment (nickchapman-da) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala]: changed to the suggested approach

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ValueEnricher.scala [hunk 2]
```diff
@@ -19,13 +19,28 @@ final class ValueEnricher(engine: Engine) {
   def enrichValue(typ: Ast.Type, value: Value[ContractId]): Result[Value[ContractId]] =
     engine.preprocessor.translateValue(typ, value).map(_.toUnnormalizedValue)
 
+  def enrichVersionedValue(
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ValueEnricher.scala:22]: looks like this and some of the other methods here could be private.

**Review by cocreature — APPROVED** [review]
Thanks!

**Review by remyhaemmerle-da — COMMENTED** [review]
Nice.

**Pull request merged**